### PR TITLE
mraa: Updated to 0.9.6 and added Java support

### PR DESCRIPTION
**Pull request by andreivasiliu2211**

### CODE REVIEW

File: recipes-devtools/mraa/mraa.inc [hunk 1]
```diff
@@ -14,12 +14,36 @@ FILES_${PN}-doc += "${datadir}/mraa/examples/"
 FILES_${PN}-dbg += "${libdir}/node_modules/mraajs/.debug/ \
                     ${PYTHON_SITEPACKAGES_DIR}/.debug/"
 
-PACKAGECONFIG ??= "python nodejs"
+# include .jar files in /usr/lib/java for 64 bit builds
+FILES_${PN}_append = "${@' ${libdir}/../lib/*' if '${MACHINE}' == 'intel-baytrail-64' else ''}"
+
+# include .so symlinks in main package
+FILES_${PN}_append = "${@' ${libdir}/../lib64/*.so' if '${MACHINE}' == 'intel-baytrail-64' else ' ${libdir}/../lib/*.so'}"
```
Comment (tripzero) [inline recipes-devtools/mraa/mraa.inc]: should this be checking for ${TARGET_ARCH} instead of ${MACHINE}?

File: recipes-devtools/mraa/mraa.inc [hunk 2]
```diff
@@ -14,12 +14,36 @@ FILES_${PN}-doc += "${datadir}/mraa/examples/"
...
+cmake_do_generate_toolchain_file_append() {
+  echo "
+set (JAVA_AWT_INCLUDE_PATH ${JAVA_HOME}/include CACHE PATH \"AWT include path\" FORCE)
+set (JAVA_AWT_LIBRARY ${JAVA_HOME}/jre/lib/amd64/libjawt.so CACHE FILEPATH \"AWT Library\" FORCE)
+set (JAVA_INCLUDE_PATH ${JAVA_HOME}/include CACHE PATH \"java include path\" FORCE)
+set (JAVA_INCLUDE_PATH2 ${JAVA_HOME}/include/linux CACHE PATH \"java include path\" FORCE)
+set (JAVA_JVM_LIBRARY ${JAVA_HOME}/jre/lib/amd64/libjvm.so CACHE FILEPATH \"path to JVM\" FORCE)
+" >> ${WORKDIR}/toolchain.cmake
+}
+
```
Comment (tripzero) [inline recipes-devtools/mraa/mraa.inc]: Should this go in upstream mraa's CMake configuration?

**Comment by andreivasiliu2211** [comment]
This recipe uses cmake.bbclass for cmake based projects. It relies on the existance of a file, toolchain.cmake, which will be used by cmake to set the options for cross-compilation. This file is created in cmake.bbclass and I append to it the options for Java. It cannot be done in mraa's Cmake configuration, because it must be done inside this file, that is generated by Yocto.

**Comment by tripzero** [comment]
@andreivasiliu2211 , okay, that makes sense.  Should it be done in upstream poky then?  The java variables are missing from cmake.bbclass.

**Comment by tripzero** [comment]
@andreivasiliu2211 do you think it's possible to update this to use mraa 0.10?

**Comment by andreivasiliu2211** [comment]
Yes. I will update it tomorrow.

**Comment by andreivasiliu2211** [comment]
Updated with TARGET_ARCH instead of MACHINE.

File: recipes-devtools/mraa/mraa_0.9.6.bb [hunk 3]
```diff
@@ -1,3 +1,3 @@
 require mraa.inc
 
-SRC_URI = "git://github.com/intel-iot-devkit/mraa.git;protocol=git;rev=d430b5c2b5b282e40bc705a28158b3d682d722f5"
+SRC_URI = "git://github.com/intel-iot-devkit/mraa.git;protocol=git;rev=d8024bb516bf2d895eaff6f92115835d7ef704a2"
```
Comment (tripzero) [inline recipes-devtools/mraa/mraa_0.9.6.bb]: is this still 0.9.6?  Should we rename file to mraa_0.10.0.bb if this rev is 0.10?

Comment (mircea) [inline recipes-devtools/mraa/mraa_0.9.6.bb]: MRAA 0.10.1 does not build in Yocto with Node & Python bindings enabled. This is the latest version that has all bindings compile in Yocto. We will push an updated recipe once a fix is out for MRAA.

**Pull request merged**